hbase-common/src/main/java/org/apache/hadoop/hbase/CellComparator.java

```diff
@@ -82,6 +82,15 @@ static CellComparator getInstance() {
    */
   int compareRows(Cell cell, byte[] bytes, int offset, int length);
 
+  /**
+   * Compares two row bytes
+   * @param leftRow the byte array of the left row
+   * @param rightRow the byte array of the right row
+   * @return greater than 0 if leftRow is bigger, less than 0 if rightRow is bigger, 0 if both
+   *         rows are equal
+   */
+  int compareRows(byte[] leftRow, byte[] rightRow);
```

Comment:
I think you should provide a default implementation to be safe compatibility-wise. See how the   ```default int compareRows(ByteBuffer row, Cell cell) {``` does it in this class.